hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -822,10 +854,21 @@ private void bindAWSClient(URI name, boolean dtEnabled) throws IOException {
         S3_CLIENT_FACTORY_IMPL, DEFAULT_S3_CLIENT_FACTORY_IMPL,
         S3ClientFactory.class);
 
+    // If there's no endpoint set, then use the default for bucket or AccessPoint. Overriding is
+    // useful when using FIPS or DualStack S3 endpoints.
+    String endpoint = conf.getTrimmed(ENDPOINT, "");
+    if (endpoint.isEmpty()) {
```

Comment:
did you meant to check for `!endpoint.isEmpty()`?

Reply:
No, what I initially intended is to say "if you're not setting the endpoint then I'll provide a default Access Point endpoint". This is because I don't know what endpoint the user wants to point it to.

This is also why your tests are failing when you set the endpoint to `ap-south-1`. I'm open to adding another `fs.s3a.accesspoint-endpoint` configuration if it's better to provide an option to override only the access point endpoint.

Reply:
So, if the endpoint isn't set/empty, then we set it to the Accesspoint endpoint, but if the endpoint is set, we set it to empty? 
Maybe set endpoint to Accesspoint endpoint if Accesspoint is set in both cases?
Maybe I got it totally wrong, please, correct me, if this isn't what we're doing.

Reply:
So, basically, if we have an Accesspoint set, we can't have an endpoint set as well?

Reply:
So I thought about this a bit more and decided to change to always use the access point endpoint instead of the above logic. So right now they should work even if you have set a custom endpoint to be something different.